tinyagentos/cluster/manager.py

```
@@ -93,22 +93,28 @@ async def register_worker(self, info: WorkerInfo) -> None:
+                    worker_hardware=info.hardware,
+                    worker_name=info.name,
+                    notifications=self._notifications,
+                )
+            except Exception:
+                logger.exception(
+                    "model_archive: promotion scan failed for worker '%s'",
+                    info.name,
+                )
+
+        import asyncio as _asyncio
+        _asyncio.create_task(_promote_bg())
```

Ruff (0.15.15)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)